app/models/deterministic_assignment_creation.rb

```diff
@@ -14,13 +14,14 @@ def self.create!(params)
   end
 
   def save!
-    if !split.feature_gate? && !existing_assignment
+    if should_create_or_update_assignment?
       ArbitraryAssignmentCreation.create!(
         visitor_id: visitor_id,
         split_name: split_name,
         variant: variant_calculator.variant,
         mixpanel_result: mixpanel_result,
-        context: context
+        context: context,
+        updated_at: updated_at
```

Comment:
Some of this coupling between `DeterministicAssignmentCreation` and `ArbitraryAssignmentCreation` makes this code hard to follow.

Since the behavior around `mixpanel_result` is specific to existing assignments, I think we can increase the clarity by isolating the changes to this class.

What we want to express is that an assignment event that occurs when there already is an assignment means that we are only updating the `mixpanel_result`.

Something like:

```ruby
if existing_assignment
  existing_assignment.assign_attributes(mixpanel_result: mixpanel_result)
  existing_assignment.save!(touch: false)
else
  ArbitraryAssignmentCreation...
end
```

I realize we paired on this, but seeing it in PR form makes me think we're introducing too much complexity.